### ISSUE TITLE
Implement LSP code action for disabling rule in configuration

### ISSUE THREAD

**Issue by anderseknert** [open-policy-agent/regal#622]
The language server should provide a [code action](https://microsoft.github.io/language-server-protocol/specifications/lsp/3.18/specification/#textDocument_codeAction) to allow developers to disable a rule in `.regal/config.yaml` via the "Quick fix" menu provided where the error is reported.

If no `.regal/config.yaml` is found, we should create one.

<img width="707" alt="Screenshot 2024-04-04 at 09 46 42" src="https://github.com/StyraInc/regal/assets/510711/c63a3f78-da9f-4c86-b58b-1c80c7cb703b">

**Comment by anderseknert** [comment]
I looked into this briefly last night, and the big challenge here is preserving user comments. These are quite common in `.regal/config.yaml` files, and they should be! That's a great way to document why some rule is disabled, or configured the way it is. Wiping these as part of a marshalling roundtrip would be terrible, so we'll need to make this a custom process... which makes it non-trivial.

**Comment by anderseknert** [comment]
Honestly, it's probably easier to just skip parsing altogether, just grep for `<category>:` and then insert text straight into the document.

**Comment by charlieegan3** [comment]
https://forum.golangbridge.org/t/modify-yaml-files-using-golang/31838/3 on the other hand looks more hopeful.

**Comment by anderseknert** [comment]
Yeah, I looked at that but didn’t find a good example like that! It’s worth a shot, although it’s not 100% lossless AFAICS, but perhaps it’s fine to mess a little with whitespace.
